admin-dev/themes/new-theme/js/pages/order/view/order-product-cancel.js

```diff
@@ -0,0 +1,143 @@
+    $(OrderViewPageMap.cancelProduct.inputs.quantity).off('change').on('change', (event) => {
+      const $productQuantityInput = $(event.target);
+      const $parentCell = $productQuantityInput.parents(OrderViewPageMap.cancelProduct.table.cell);
+      const $productAmount = $parentCell.find(OrderViewPageMap.cancelProduct.inputs.amount);
+      const productQuantity = parseInt($productQuantityInput.val(), 10);
+      if (productQuantity <= 0) {
+        $productAmount.val(0);
+        this.updateVoucherRefund();
+
+        return;
+      }
+      const priceFieldName = this.isTaxIncluded ? 'productPriceTaxIncl' : 'productPriceTaxExcl';
+      const productUnitPrice = parseFloat($productQuantityInput.data(priceFieldName));
+      const amountRefundable = parseFloat($productQuantityInput.data('amountRefundable'));
+      const guessedAmount = productUnitPrice * productQuantity < amountRefundable ?
+        productUnitPrice * productQuantity : amountRefundable;
+      const amountValue = parseFloat($productAmount.val());
+      if ($productAmount.val() === '' || amountValue === 0 || amountValue > guessedAmount) {
+        $productAmount.val(guessedAmount);
+        this.updateVoucherRefund();
+      }
+    });
+
+    $(OrderViewPageMap.cancelProduct.inputs.amount).off('change').on('change', () => {
+      this.updateVoucherRefund();
+    });
+  }
+
+  updateVoucherRefund() {
+    let totalAmount = 0;
+    $(OrderViewPageMap.cancelProduct.inputs.amount).each((index, amount) => {
+      const floatValue = parseFloat(amount.value);
+      totalAmount += !isNaN(floatValue) ? floatValue : 0;
+    });
+
+    this.updateVoucherRefundTypeLabel($(OrderViewPageMap.cancelProduct.radios.voucherRefundType.productPrices), totalAmount);
+    const refundVoucherExcluded = totalAmount - this.discountsAmount;
+    this.updateVoucherRefundTypeLabel($(OrderViewPageMap.cancelProduct.radios.voucherRefundType.productPricesVoucherExcluded), refundVoucherExcluded);
+    if (refundVoucherExcluded < 0) {
+      $(OrderViewPageMap.cancelProduct.radios.voucherRefundType.productPricesVoucherExcluded).prop('checked', false).prop('disabled', true);
+      $(OrderViewPageMap.cancelProduct.radios.voucherRefundType.productPrices).prop('checked', true);
+      $(OrderViewPageMap.cancelProduct.radios.voucherRefundType.negativeErrorMessage).show();
+    } else {
+      $(OrderViewPageMap.cancelProduct.radios.voucherRefundType.productPricesVoucherExcluded).prop('disabled', false);
+      $(OrderViewPageMap.cancelProduct.radios.voucherRefundType.negativeErrorMessage).hide();
+    }
+  }
+
+  updateVoucherRefundTypeLabel($input, refundAmount) {
+    const defaultLabel = $input.data('defaultLabel');
+    const $label = $input.parents('label');
+    const formattedAmount = this.currencyFormatter.format(refundAmount);
+
+    // Change the ending text part only to avoid removing the input
+    $label.get(0).lastChild.nodeValue = `
+    ${defaultLabel} ${formattedAmount}`;
```

Comment:
Is line break wanted?

Reply:
Yep it's on purpose, this is how the twig widget is displayed It avoids to text to be too close to the radio

Reply:
Could you add a little comment, look like it's a missclick :sweat_smile: